src/Aquifer.Common/Services/Caching/CachingVersificationService.cs

```diff
@@ -1,108 +1,216 @@
 using System.Collections.ObjectModel;
 using Aquifer.Data;
+using Aquifer.Data.Enums;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Caching.Memory;
 
 namespace Aquifer.Common.Services.Caching;
 
 public interface ICachingVersificationService
 {
-    Task<ReadOnlyDictionary<int, int>> GetBaseVerseIdByBibleVerseIdMapAsync(int bibleId, CancellationToken cancellationToken);
+    Task<ReadOnlyDictionary<int, string>> GetBaseVerseIdWithOptionalPartByBibleVerseIdMapAsync(int bibleId, CancellationToken cancellationToken);
```

Comment:
Updated to `string` to support base verse parts like `1002003002a` as a value (or key for the inverted dictionary).